src/lib.rs

```diff
@@ -8,35 +8,270 @@ mod transaction;
 pub mod update;
 
 use crate::{
-    keys::{OwnershipKeyPair, OwnershipPublicKey, PublishingKeyPair, RevocationKeyPair},
+    create::{BuildFundingPSBT, SignFundingPSBT},
+    keys::{
+        OwnershipKeyPair, OwnershipPublicKey, PublishingKeyPair, PublishingPublicKey,
+        RevocationKeyPair, RevocationPublicKey, RevocationSecretKey,
+    },
     transaction::{CommitTransaction, FundingTransaction, SplitTransaction},
 };
 use anyhow::bail;
-use bitcoin::Amount;
+use bitcoin::{Amount, Transaction};
 use ecdsa_fun::adaptor::EncryptedSignature;
-use keys::{PublishingPublicKey, RevocationPublicKey, RevocationSecretKey};
+
+#[allow(dead_code)]
+pub struct Channel {
+    x_self: OwnershipKeyPair,
+    X_other: OwnershipPublicKey,
+    pub TX_f_body: FundingTransaction,
+    pub current_state: ChannelState,
+    revoked_states: Vec<RevokedState>,
+}
+
+#[async_trait::async_trait]
+pub trait BroadcastSignedTransaction {
+    async fn broadcast_signed_transaction(&self, transaction: Transaction) -> anyhow::Result<()>;
+}
+
+#[async_trait::async_trait]
+pub trait SendMessage {
```

Comment:
I think one pair of `SendMessage` and `ReceiveMessage` pair per protocol is going to be a pain. That's 6 trait implementations that are going to look the same.

I propose to have one pair at crate level and one `enum Message` that wraps **all** possible messages.

Reply:
> one enum Message that wraps all possible messages

Rather than wrapping all possible messages, which doesn't offer much type safety, what if we just send around `String` and implement `Serialize` and `Deserialize` on all the messages?

Reply:
Going for this now as it's simpler to implement and I don't want to think about ser/deser right now.

Reply:
> Rather than wrapping all possible messages, which doesn't offer much type safety, what if we just send around String and implement Serialize and Deserialize on all the messages?

Because the lib consumer may want to use a binary format and we are hijacking them by forcing a first string serialisation.

Ideally we should implement `Serialize` and `Deserialize` for all our messages but feature gate it behind a `serde` flag. That would allow a lib consumer to use serde out of the box or not use it and do their own serialisation/deserialisation of the messages.

Finally, in term of type safety, it is not really needed there, we just need a `fn send` feature that sends messages.

The stronger type safety should be around what is returned by `fn receive`.